core/jobs/batch_jobs/audit_rule_validation_jobs.py

```diff
@@ -0,0 +1,1004 @@
+        filter_invalid_drag_drop_rules = (
+            combine_exp_ids_and_states
+            | 'Get invalid drag drop rules' >> beam.MapTuple(
+                lambda exp_id, exp_states, exp_created_on: (
+                    exp_id,
+                    self.invalid_drag_drop_interactions(exp_states),
+                    exp_created_on.date()
+                )
+            )
+            | 'Remove empty values drag drop rule' >> beam.Filter(
+                lambda exp: len(exp[1]) > 0)
+        )
+
+        report_count_invalid_drag_drop_rules = (
+            filter_invalid_drag_drop_rules
+            | 'Report count for invalid drag drop rules' >> (
+                job_result_transforms.CountObjectsToJobRunResult(
+                    'NUMBER OF EXPS WITH INVALID DRAG DROP RULES')
+            )
+        )
+
+        report_invalid_drag_drop_rules = (
+            filter_invalid_drag_drop_rules
+            | 'Show info for drag drop' >> beam.MapTuple(
+                lambda exp_id, exp_drag_errors, exp_created_on: (
+                    job_run_result.JobRunResult.as_stderr(
+                       f'The id of exp is {exp_id}, '
+                       f'created on {exp_created_on}, and the invalid '
+                       f'drag drop states are {exp_drag_errors}'
+                    )
+                )
+            )
+        )
+
+        # DragAndDrop invalid curated rules.
+        filter_invalid_curated_drag_drop_rules = (
+            curated_explorations
+            | 'Get invalid curated drag drop rules' >> beam.MapTuple(
+                lambda exp_id, exp_states, exp_created_on: (
+                    exp_id,
+                    self.invalid_drag_drop_interactions(exp_states),
+                    exp_created_on.date()
+                )
+            )
+            | 'Remove empty values curated drag drop rule' >> beam.Filter(
+                lambda exp: len(exp[1]) > 0)
+        )
+
+        report_count_invalid_curated_drag_drop_rules = (
+            filter_invalid_curated_drag_drop_rules
+            | 'Report count for invalid curated drag drop rules' >> (
+                job_result_transforms.CountObjectsToJobRunResult(
+                    'NUMBER OF EXPS WITH INVALID CURATED DRAG DROP RULES')
+            )
+        )
+
+        report_invalid_curated_drag_drop_rules = (
+            filter_invalid_curated_drag_drop_rules
+            | 'Show info for curated drag drop' >> beam.MapTuple(
+                lambda exp_id, exp_drag_errors, exp_created_on: (
+                    job_run_result.JobRunResult.as_stderr(
+                       f'The id of exp is {exp_id}, '
+                       f'created on {exp_created_on}, and the invalid '
+                       f'curated drag drop states are {exp_drag_errors}'
+                    )
+                )
+            )
+        )
+
+        # Continue Text should be non-empty and have a max-length of 20.
+        filter_invalid_continue_text_values = (
```

Comment:
This should somehow convey that it contains the language code.

Reply:
Done